packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmConfigurationImpl.kt

```diff
@@ -75,7 +75,7 @@ open class RealmConfigurationImpl constructor(
         } else path
         this.name = name // FIXME Should read name from end of path
         this.schema = schema
-        this.mapOfKClassWithCompanion = companionMap
+        this.mapOfKClassWithCompanion = schema.map { it to it.realmObjectCompanion()!! }.toMap()
```

Comment:
Wouldn't it be better to handle the null pointer exception in the `realmObjectCompanion()` methods? If, for some reason, there is no realmObjectCompanion we get a rather unhelpful NullPointerException here.

If we check inside the method we can also craft a better error message containing the name of the offending class.

Reply:
👍  Makes sense